tests/unit/dbService.js

```diff
@@ -0,0 +1,361 @@
+'use strict'
+
+const tap = require('tap')
+
+const db = require('../../src/db')
+
+const {
+  createEndpoint,
+  deleteEndpoint,
+  deleteEndpoints,
+  readEndpoints,
+  updateEndpoint
+} = require('../../src/db/services/endpoints')
+const EndpointModel = require('../../src/db/models/endpoints')
+
+db.open('mongodb://localhost:27017/unitTest')
+
+tap.test('Database interactions', {autoend: true}, t => {
+  t.test('CRUD interactions with DB', {autoend: true}, async t => {
+    t.beforeEach(async () => {
+      await EndpointModel.deleteMany({})
+      console.debug('Deleted all resources in DB')
+    })
+
+    t.tearDown(async () => {
+      await db.close()
+      console.debug('Closed all connections to DB')
+    })
+
+    t.test(
+      'should fail to create Endpoint when validation fails',
+      {autoend: true},
+      async t => {
+        t.plan(1)
+        const testEndpointConfig = {
+          name: 'test',
+          endpoint: {
+            pattern: '/test'
+          },
+          transformation: {
+            input: 'JSON',
+            output: null
+          }
+        }
+        await createEndpoint(testEndpointConfig)
+          .then(data => {
+            t.fail(`Should not reach here. ${data}`)
+          })
+          .catch(error => {
+            t.equals(
+              error.message,
+              'endpoint validation failed: transformation.output: Path `transformation.output` is required.'
+            )
+          })
+      }
+    )
+
```

Comment:
Maybe we should add in another test here that verifies if a endpoint already exists with name/pattern that we cant create an endpoint with those same details.

This could be done as part of the updated tests story